hie-base/Haskell/Ide/Engine/PluginTypes.hs

```diff
@@ -280,8 +282,15 @@ instance Eq ParamValP where
  (ParamPosP  p) == (ParamPosP p') = p == p'
  _ == _ = False
 
+-- pattern ParamTextP :: () =>
```

Comment:
It should actually be uncommented. I was getting warnings on missing signatures, which led to https://ghc.haskell.org/trac/ghc/ticket/12484